### ISSUE TITLE
Inserter further improvements

### ISSUE THREAD

**Issue by afercia** [WordPress/gutenberg#3974]
Follow up to #1823 

During WCUS contributor day, some Gutenberg features were tested in depth, big thanks to Jackie D’Elia (@jdelia) and Katherine White (@katwhite). Full report: https://gist.github.com/kwhite/22cf005d0a23f3c66dd8df62a6bde574

Specifically for the Inserter: navigating with the keyboard through the categories group is sometimes difficult and/or confusing. The main friction points:

1.
> The down arrow should also take you into the following Section, ideally. I know that sucks and it’s hard to do, but it would really help.

I'd agree sometimes you just feel "stuck" and you don't know how to navigate to the following group, especially because it can be out of view. Making the arrows navigate also to the next section could mitigate this issue.

2.
> Keep the convenience of tabbing between Sections within a Tab. That’s great and really useful.

Yay!

3.
> However, it’s currently disorienting when tabbing between Sections within a Tab; rather than being taken back to the last Item I had highlighted, I’d rather be taken to the top of the Section so I know where I am.

See also 4:

4.
> The Section title needs to be visible when you are tabbing back UP through the Sections. Right now it ends up hidden by the Tab navigation bar.

To me, it seems that point 3 and 4 are strictly related. When jumping to a new category group with the Tab key, focus goes to the last item you had focused in that group (it's the "roving tabindex" thing). In this situation, you may end up not being able to understand where you are because the category group title is not visible, for example:

![screen shot 2017-12-13 at 15 43 59](https://user-images.githubusercontent.com/1682452/33944444-7ed58964-e01c-11e7-95a6-ae0a1fdb0417.png)

So I'd say to consider to:
- either always focus the first item in a group
- or make the group title always visible: e.g. the emoji inserter on macOS (Ctrl-Cmd-Spacebar to invoke it)

5.
> The right/left arrows need to work as well as the up/down ones for navigating between items within a section. If they are visually in two columns, there will be some users who intuitively think of them as right/left based.

I think this is already in the works?

6.
> Embeds section: Using Section labels would be helpful here; it looks like we use the most popular embeds at the top and then show the rest in alphabetical order. Splitting these out with clear labels would be helpful.

Good catch! Haven't noticed before that after Vimeo then the embeds are listed alphabetically starting with Animoto. Splitting in two section could help

7.
> Embeds section: "Speaker" Item in this section... is this SpeakerDeck? I ask b/c it worked with a SpeakerDeck embed; but if that's the case, the name should be updated.

**Comment by SuperGeniusZeb** [comment]
The inserter has been reworked a lot since this issue was made. Are the problems stated here still relevant or should this issue be closed/replaced?

**Comment by afercia** [comment]
Thanks for the reminder 🙂 Yep, so many things have changed, I'd say this can be closed.
